TW-594 Collections and workspaces on private API network (#4268)
postmanlabs/postman-docs: src/pages/docs/collaborating-in-postman/adding-private-network.md

  - type: link
    name: "ChargeHub"
    url: "https://www.postman.com/case-studies/chargehub/"
    search_keyword: "Private API Network, API sharing, folders, network listing, filtering apis, publish versions, github import, private apis, adding apis"

---

> 

Reviewer: IIRC, this feature is Enterprise-only going forward? If so we should probably have a note to that effect at the top of the page.

    * [Removing elements from the Private API Network](#removing-elements-from-the-private-api-network)
* [Private API Network reports](#private-api-network-reports)

## Navigating the Private API Network

The Private API Network is a good place to learn about workspaces, collections, and APIs shared within your team. Under your team name, you can browse a directory of elements shared within your team.

There are two ways to access the Private API Network:

* Select **Home** from the Postman header, then select **Private API Network** in your team information on the left side.
* Select **API Network** from the Postman header, then select **Private API Network**.

In the Private API Network, you can filter elements by name using the search box. Select **Added by** to filter based on the person who added the element. Select **Type** to filter by the type of element. Select **Sort by** to sort elements based on name and date added. You can also filter folders, sub-folders, and elements based on name and date added.

<img alt="Private API List" src="https://assets.postman.com/postman-docs/v10/private-api-network-list-v10.jpg"/>

> To review information about an element, select it from the list. You can view the element's description and the editors who have worked on it. For workspaces, you can view all of the collections and APIs that reside in them. For collections, you can view available documentation. For APIs, you can view definitions and associated collections.

Reviewer: ```suggestion
To review information about an element, select it from the list. You can view the element's description and the editors who have worked on it. For workspaces, you can view all of the collections and APIs inside them. For collections, you can view available documentation. For APIs, you can view definitions and associated collections.
```

* **If your team doesn't use the optional approval process**, any user with an Editor role for an element can add it to the Private API Network.

### Adding workspaces

If your team uses the [optional approval process](#using-the-approval-process-workflow), only an API Network Manager can add workspaces directly to the Private API Network. Workspace Editors must [request to add a workspace](#requesting-to-add-an-api). If your team doesn't use the approval process, any user with Editor access for the workspace can add it to the Private API Network.

To add a workspace to the Private API Network from the API Builder:

1. Open the workspace you want to add to the Private API Network.
1. In the workspace overview, select **Add to API Network**.
1. (Optional) Select a folder or create one to keep elements organized.
1. Select **Add**.

> You can also add a workspace while in your team's Private API Network:

Reviewer: Maybe: 
```suggestion
You can also add a workspace from inside your team's Private API Network:
```

❓

1. Open your [Private API Network](https://go.postman.co/network/private).

> 1. Select **Add to network**

Reviewer: ```suggestion
1. Select **Add to network**.
```

1. Select **Add to network**
1. Search for and select the collection you want to add.
1. (Optional) Select a folder or create one to keep elements organized.
1. Select **Add**.

When you add a collection to the [Private API Network](https://go.postman.co/network/private), it's visible to your Postman team, but isn't visible to [partners](/docs/collaborating-in-postman/using-workspaces/partner-workspaces/).

The collections that you add to the Private API Network reflect the latest state of the collection in your team workspace. In other words, changes made to the collection in the workspace are reflected in the network in real time.

### Adding APIs

If your team uses the [optional approval process](#using-the-approval-process-workflow), only an API Network Manager can add APIs directly to the Private API Network. API Editors must [request to add an API](#requesting-to-add-an-api). If your team doesn't use the approval process, any user with Editor access for the API can add it to the Private API Network.

To add an API to the Private API Network from the API Builder:

1. Open the API you want to add to the Private API Network.

> 1. In the API overview, select **Add to API Network**.

Reviewer 2: I think this changed to be just Publish API, but verify.

### Adding APIs

If your team uses the [optional approval process](#using-the-approval-process-workflow), only an API Network Manager can add APIs directly to the Private API Network. API Editors must [request to add an API](#requesting-to-add-an-api). If your team doesn't use the approval process, any user with Editor access for the API can add it to the Private API Network.

To add an API to the Private API Network from the API Builder:

1. Open the API you want to add to the Private API Network.
1. In the API overview, select **Publish API**.
1. Select **Request to add to Private API Network**.
1. (Optional) Select a folder or create one to keep elements organized.
1. Select **Add**.

You can also add an API while in your team's Private API Network:

1. Open your [Private API Network](https://go.postman.co/network/private).
1. Select **Add to network**
1. Search for and select the API you want to add.
1. (Optional) Select a folder or create one to keep elements organized.
1. Select **Add**.

When you add an API to the [Private API Network](https://go.postman.co/network/private), it's visible to your Postman team.

> The APIs that you add to the Private API Network reflect the latest state of the API in your team workspace. In other words, changes made to the API in the workspace are reflected in the network in real time.

Reviewer 3: we should probably denote that published changes are reflected, not just changes alone.

1. Select **Consumers can view this version**.
1. Select **Save Changes**.

> #### Importing APIs from code repositories

Reviewer 2: We have competing edits between this and TW-493, with the latter one being correct.

* **If your team doesn't use the optional approval process**, any user with an Editor role can complete these tasks.

### Organizing with folders

The sidebar navigation displays the folder structure for your Private API Network. You can drag elements and sub-folders into different folders. You can also add descriptions to folders to describe elements within the folders.

<img alt="Create new folder in Private Network" src="https://assets.postman.com/postman-docs/v10/private-api-network-create-folder-in-sidebar-v10.jpg" width="300px"/>

To create a new folder from the Private API Network view:

1. Select __Create Folder__ from the sidebar.
1. Give the folder a name.
1. (Optional) Give a description.
1. Select **Save**.

To create a new folder from the Private API Network overview page:

1. Select **Create Folder** on the right.
1. Give the folder a name.
1. (Optional) Give a description.
1. Select **Save**.

You can also use **Create folder** to create sub-folders inside a folder.

Use **Search elements and folders** to search across folders, sub-folders, and elements in your Private API Network.

### Editing element listings

> Select the more actions icon <img alt="More actions icon" src="https://assets.postman.com/postman-docs/icon-more-actions-v9.jpg#icon" width="16px"> next to the element you would like to edit from the network, then select **Edit element** to change its location. Select **Edit** to save your changes.

Reviewer 3: We should also add that for collections **Edit element** allows you to edit the summary, and associated environments.
